command.go

```diff
@@ -57,6 +57,10 @@ type Command struct {
 	// Example: add [-F file | -D dir]... [-f format] profile
 	Use string
 
+	// CommandName is the command name if set. Otherwise the command
+	// name is the first word of the `Use` line.
+	CommandName string
```

Comment:
Yes, this solves special case for the root command. Maybe it is best to document this and ignore on other commands, with a warning (if we have can) about "ignoring CommandName for non-root command".

Maybe we can use an annotation instead of adding new name to the struct?

I'll fix the spelling in the next version.

Reply:
My first reaction is that I like the use of an annotation.
With a more specific solution only affecting the help, and an annotation to specify the name to be used by the help, we may have a cleaner solution.

I'll let you give it a go and see how you feel about it.

Reply:
I think the current version is better - it is little harder to use, but this is a very special case so it is better to have a special solution and keep the general command interface cleaner.